src/FastConv/FastConv.cpp

```diff
@@ -1,7 +1,256 @@
+
+        CFft::createInstance(pcFFT);
+        pcFFT->initInstance(2 * m_iBlockLength, 1, CFft::kWindowHann, CFft::kNoWindow);
+
+        pfComplexTemp = new CFft::complex_t[2 * m_iBlockLength];
+
+        pfFFTRealTemp = new float[m_iBlockLength + 1]; 
+        pfFFTImagTemp = new float[m_iBlockLength + 1];
+        pfCurrentBlockFFTReal = new float[m_iBlockLength + 1];
+        pfCurrentBlockFFTImag = new float[m_iBlockLength + 1];
+        pfIFFTTemp = new float[2 * m_iBlockLength]{0};
+
+        m_ppfIRFreqDomainReal = new float* [m_iBlockNum];
+        m_ppfIRFreqDomainImag = new float* [m_iBlockNum];
+        m_pfInputBlockBuffer = new float[2 * m_iBlockLength]{ 0 };
+        m_ppfProcessedBlockBuffer = new float* [m_iBlockNum];
+        for (int i = 0; i < m_iBlockNum; i++)
+        {
+            m_ppfIRFreqDomainReal[i] = new float[m_iBlockLength + 1]{ 0 };
+            m_ppfIRFreqDomainImag[i] = new float[m_iBlockLength + 1]{ 0 };
+            m_ppfProcessedBlockBuffer[i] = new float[m_iBlockLength] {0};
+            for (int j = 0; j < m_iBlockLength; j++)
+            {
+                if (i * iBlockLength + j < m_iLengthOfIr)
+                    pfIFFTTemp[j] = pfImpulseResponse[i * iBlockLength + j];
+                else
+                    pfIFFTTemp[j] = 0;
+            }
+
+            for (int j = m_iBlockLength; j < 2 * m_iBlockLength; j++)
+                pfIFFTTemp[j] = 0;
+
+            pcFFT->doFft(pfComplexTemp, pfIFFTTemp);
+            pcFFT->splitRealImag(m_ppfIRFreqDomainReal[i], m_ppfIRFreqDomainImag[i], pfComplexTemp);
+        }
+    }
+
+    virtual ~CConvFreqDomain()
+    {
+        for (int i = 0; i < m_iBlockNum; i++)
+        {
+            delete[] m_ppfProcessedBlockBuffer[i];
+            delete[] m_ppfIRFreqDomainReal[i];
+            delete[] m_ppfIRFreqDomainImag[i];
+        }
+        delete[] m_ppfIRFreqDomainReal;
+        delete[] m_ppfIRFreqDomainImag;
+        delete[] m_pfInputBlockBuffer;
+        delete[] m_ppfProcessedBlockBuffer;
+
+        m_ppfIRFreqDomainReal = 0;
+        m_ppfIRFreqDomainImag = 0;
+        m_pfInputBlockBuffer = 0;
+        m_ppfProcessedBlockBuffer = 0;
+
+        delete[] pfIFFTTemp;
+        delete[] pfFFTRealTemp;
+        delete[] pfFFTImagTemp;
+        delete[] pfCurrentBlockFFTReal;
+        delete[] pfCurrentBlockFFTImag;
+        pfIFFTTemp = 0;
+        pfFFTRealTemp = 0;
+        pfFFTImagTemp = 0;
+
+
+        delete[] pfComplexTemp;
+        pfComplexTemp = 0;
+
+        CFft::destroyInstance(pcFFT);
+        pcFFT = 0;
+    }
+
+    void process(float* pfOutputBuffer, const float* pfInputBuffer, int iLengthOfBuffers)
+    {
+
+        for (int i = 0; i < iLengthOfBuffers; i++)
+        {
+
+            assert(m_iWriteIdx + m_iBlockLength < 2 * m_iBlockLength);
+            assert(m_iReadBlockIdx < m_iBlockNum);
+            assert(m_iWriteBlockIdx < m_iBlockNum);
+
+
+            m_pfInputBlockBuffer[m_iWriteIdx + m_iBlockLength] = pfInputBuffer[i];
```

Comment:
m_pfInputBlockBuffer starts to accept samples at index m_iBlockLength, half the size of m_pfInputBlockBuffer. Is it overlap-and-save? Just curious.

Reply:
Yes it is!